lib/commands/query.js

```diff
@@ -64,7 +65,17 @@ class Query extends BaseCommand {
       forceActual: true,
     }
     const arb = new Arborist(opts)
-    const tree = await arb.loadActual(opts)
+    let tree
+    if (this.npm.config.get('package-lock-only')) {
+      try {
+        tree = await arb.loadVirtual()
+      } catch (err) {
+        /* eslint-disable-next-line max-len */
+        throw this.usageError('A package lock or shrinkwrap file is required in package-lock-only mode')
```

Comment:
Are there other potential errors that `loadVirtual` could throw here? Can we detect `ENOENT` or similar and only throw the usageError on that?

Reply:
This mirrors the error elsewhere in npm itself. If that's a problem it exists beyond the scope of this PR.

Reply:
Check out the `ci.js` command.

Reply:
makes sense. can we add a `log.verbose('loadVirtual', er.stack)` to match what ci.js does?

Reply:
done